section3/reflection.md

```diff
@@ -1,13 +1,19 @@
 ## Section 3 Reflection
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
+- At first I was intimidated and was doubting myself a little bit.  One thing I have learned through this project and Mod0 is to take everything a day at a time.  Easier said than done, but it has helped me persevere through problem solving.  I believe that researching and utilizing resources helps me learn and grasp the material even more.
 
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
+- I use Apple notes on a daily basis and this an array because it stores a LIST of values and all of the values are of different varieties.  For example, my grocery list: 2 apples, or onion.
 
 3. In your own words, explain what an Array method is.
+- An array method is essentially a function that we can apply to an array.  They are unique in that they modify or change an array.  For example, the push() method will add an element to the end of an array - then we do not need to retype the original.
 
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+- A loop will check a condition a certain number of times and as long as the condition is true, it will continue to check.  The loop will end when it finds the condition to be false.
 
 5. If you had to teach a 5 year old about loops, how would you explain it?
+- I have a cookie jar and I made 10 cookies.  I need to put them into the cookie jar but only one at a time.  I pick up one cookie, I put one cookie in.  I pick up the second cookie, I put the second cookie in.  I pick up the third cookie, I put the third cookie in.  You keep doing this until all of your cookies are in the jar.  Once all ten are inside, you close the lid and you are finished.
 
 6. What questions do you still have about Arrays and loops?
+- I have all the questions!  This is not related to arrays and loops but is there a standard practice for double nesting - as in is it common to do or are there other methods that are way less chaotic and nicer to look at?
```

Comment:
Double nesting definitely does happen, but it is nice to avoid it in production applications when possible since that slows down the application.